src/ops.rs

```diff
@@ -0,0 +1,411 @@
+use crate::array::ArrayElement;
+use crate::{array::Array, stream::StreamOrDevice};
+use num_traits::NumCast;
+
+impl Array {
+    /// Construct an array of zeros.
+    ///
+    /// Example:
```

Comment:
I think the rust convention for example usually has # before Example

```rust
/// # Example
/// ```rust
/// foo();
/// ```
fn foo() { }

```